src/ParticleEmitter_TEST.cc

```diff
@@ -30,37 +31,139 @@
 using namespace ignition;
 using namespace rendering;
 
+/// \brief The test fixture.
 class ParticleEmitterTest : public testing::Test,
                             public testing::WithParamInterface<const char *>
 {
-  /// \brief Test basic API
-  public: void ParticleEmitter(const std::string &_renderEngine);
+  /// \brief How to set up the environment.
+  public: bool SetUp(const std::string &_renderEngine);
+
+  /// \brief Check setters/getters.
+  public: void CheckBasicAPI();
+
+  /// \brief How to tear down the environment.
+  public: void TearDown();
+
+  /// \brief The rendering engine used.
+  protected: RenderEngine *engine;
+
+  /// \brief The scene.
+  protected: ScenePtr scene;
 };
 
 /////////////////////////////////////////////////
-void ParticleEmitterTest::ParticleEmitter(const std::string &_renderEngine)
+bool ParticleEmitterTest::SetUp(const std::string &_renderEngine)
 {
-  RenderEngine *engine = rendering::engine(_renderEngine);
-  if (!engine)
+  this->engine = rendering::engine(_renderEngine);
+  if (!this->engine)
   {
     igndbg << "Engine '" << _renderEngine << "' is not supported" << std::endl;
-    return;
+    return false;
   }
 
-  ScenePtr scene = engine->CreateScene("scene");
+  this->scene = this->engine->CreateScene("scene");
+  return this->scene != nullptr;
+}
+
+/////////////////////////////////////////////////
+void ParticleEmitterTest::CheckBasicAPI()
+{
+  // Create a particle emitter.
+  ParticleEmitterPtr particleEmitter = this->scene->CreateParticleEmitter();
+
+  // Default values.
+  EmitterType              expectedEmitterType  = EmitterType::EM_POINT;
+  ignition::math::Vector3d expectedEmitterSize  = ignition::math::Vector3d::One;
+  double                   expectedRate         = 10.0;
+  ignition::math::Pose3d   expectedLocalPose    = ignition::math::Pose3d::Zero;
+  double                   expectedDuration     = 0;
+  bool                     expectedEmitting     = false;
+  ignition::math::Vector3d expectedParticleSize = {100, 100, 100};
+  double                   expectedLifetime     = 5;
+  MaterialPtr              expectedMaterial     = nullptr;
+  double                   expectedMinVel       = 1;
+  double                   expectedMaxVel       = 1;
+  ignition::math::Color    expectedColorStart   = ignition::math::Color::White;
+  ignition::math::Color    expectedColorEnd     = ignition::math::Color::White;
+
+  // Check default expectations.
+  EXPECT_EQ(       expectedEmitterType,  particleEmitter->Type());
```

Comment:
the nice formatting here produced a few code check errors for me :/

```
./src/ParticleEmitter_TEST.cc:90:  Extra space after ( in function call  [whitespace/parens] [4]
```

Reply:
I don't see any errors on cppcheck `1.82` but I tweaked it a bit. If you still see errors feel free to edit it directly until it's clean. 1843624

Reply:
cool errors are gone now. I think it was found by [cppylint.py](https://github.com/ignitionrobotics/ign-rendering/blob/master/tools/cpplint.py#L2529) in `tools` dir when I ran `sh tools/code_check.sh` locally.